eth_trie/nibbles.nim

```diff
@@ -99,6 +99,61 @@ proc hexPrefixDecode*(r: ByteRange): tuple[isLeaf: bool, nibbles: NibblesRange]
   else:
     result.isLeaf = false
 
+template putNibble(bytes, x: untyped) =
```

Comment:
There is another similar proc in the module called `writeNibbles`. The definition is a bit more complicated only because it iterates between `ibegin` and `iend` (which are absolute positions) and then uses the `{}` operator which provides absolute indexing. It can be split in two just like `putNibbles` and `putNibble` and it can gain an `outputBytes` parameter.